optimum/habana/transformers/trainer.py

```diff
@@ -671,7 +663,7 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
 
         # set_to_none is not implemented for some optimizers
         try:
-            model.zero_grad(set_to_none=True)
+            model.zero_grad(set_to_none=False)
```

Comment:
`set_to_none=True` is usually recommended, we should use `set_to_none=False` only if using HPU graphs then.